cmake/options.cmake

```diff
@@ -0,0 +1,322 @@
+include(CMakeDependentOption)
+
+#------------------------------------------------------------------------------
+# Add an option with optional dependencies.
+#
+# Arguments:
+#   <DEFAULT_STATE> - Is the option enabled by default? (`ON` or `OFF`)
+#
+#   DEPENDS <expression>
+#     A list of expressions which must evaluate to true for the component to
+#     be made available. (Otherwise, the component will not be enabled, and the
+#     option for the component will be hidden.)
+#
+# Extra arguments are combined (with a single space) to form the description of
+# the option.
+#------------------------------------------------------------------------------
+function(drake_option NAME DEFAULT_STATE)
+  # "Fix" escaping so cmake_parse_arguments will split properly (note:
+  # cmake_parse_arguments splits twice, so escape two more levels).
+  string(REPLACE "\\" "\\\\\\" _args "${ARGN}")
+
```

Comment:
The `\` layers that `${ARGN}` evaluation and `cmake_parse_arguments` removes are only when they precede `;`.  Other backslashes are not removed by those steps.  If the goal here is to preserve `;`s in values from callers then this should only add backslashes before `;`s.

Until `cmake_parse_arguments(PARSE_ARGV)` is available one could implement a loop over `ARGV#` here to collect the arguments, add needed escapes on `;`s, and then call `cmake_parse_arguments`.  That would avoid the need for callers to escape anything.